src/NServiceBus.Msmq.AcceptanceTests/When_TimeToBeReceived_set_and_native_receivetransaction.cs

```diff
@@ -43,6 +45,7 @@ public class MyMessageHandler : IHandleMessages<MyMessage>
 
                 public async Task Handle(MyMessage message, IMessageHandlerContext context)
                 {
+                    Context.HandlerInvoked = true;
```

Comment:
I think we are lucky. Then because almost all transport wait a certain period of time until pending receives are done, we are probably fine. What could happen is that condition for shutting down is reached, ATT framework shuts down, receive operation is still pending, NSB waits for it to complete, SendLocal is reached and throws. So I think we are good.